src/datachain/lib/dc.py

```diff
@@ -831,8 +831,27 @@ def mutate(self, **kwargs) -> "Self":
         )
         ```
         """
-        chain = super().mutate(**kwargs)
-        chain.signals_schema = self.signals_schema.mutate(kwargs)
+        mutated = {}
+        schema = self.signals_schema
+        for name, value in kwargs.items():
+            if isinstance(value, Column) and schema.has_object(value.name):
+                # renaming existing object
+                old_signals = schema.db_signals(obj_name=value.name, as_columns=True)
+                for signal in old_signals:
+                    mutated[signal.name.replace(value.name, name, 1)] = signal
+            elif isinstance(value, Column) and value.name in schema.db_signals():
+                # renaming existing non object signal
```

Comment:
What is a use case for non-object signals? Let's get rid of theses.

Reply:
Refactored so there is only one `if` clause